src/server/lib/idapi/invertOptOutConsents.ts

```diff
@@ -0,0 +1,39 @@
+import { Consent } from '@/shared/model/Consent';
+
+/**
+ * See PR https://github.com/guardian/identity/pull/2079 for more details why this is needed.
+ * This is horrible but necessary due to business requirements because we cannot change
+ * the consent IDs "eg. Profiling_Optout" on the server data model or downstream integrations
+ */
+
+type ConversionString = '_optout' | '_optin';
+
+const invertConsents =
+  (input: ConversionString, output: ConversionString) =>
+  (pageConsents: Consent[]) => {
+    const invertedConsents = pageConsents.map((consent) => {
+      if (consent.id.includes(input)) {
+        const invertedConsent = {
+          id: consent.id.replace(input, output),
+          description: consent.description,
+          name: consent.name,
+        } as Consent;
+
+        if (consent.hasOwnProperty('consented')) {
+          return {
+            consented: !consent.consented,
+            ...invertedConsent,
+          } as Consent;
+        } else {
+          return invertedConsent;
+        }
+      } else {
+        return consent;
+      }
+    });
+
+    return invertedConsents;
+  };
```

Comment:
Use of the early return pattern could remove a significant amount of nesting here.

Reply:
Here's what I mean (not tested):

```js
const invertConsents =
  (input: ConversionString, output: ConversionString) =>
  (pageConsents: Consent[]) => {
    const invertedConsents = pageConsents.map((consent) => {
      if (!consent.id.includes(input)) {
        return consent;
      }

      const invertedConsent = {
        id: consent.id.replace(input, output),
        description: consent.description,
        name: consent.name,
      } as Consent;

      if (!consent.hasOwnProperty('consented')) {
        return invertedConsent;
      }
      
      return {
        consented: !consent.consented,
        ...invertedConsent,
      } as Consent;
    });

    return invertedConsents;
  };
```